uxarray/grid.py

```diff
@@ -371,3 +372,111 @@ def integrate(self, var_ds, quadrature_rule="triangular", order=4):
         integral = np.dot(face_areas, face_vals)
 
         return integral
+
+    def _populate_cartesian_xyz_coord(self):
+        """A helper function that populates the xyz attribute in UXarray.ds.
+        This function is called when we need to use the cartesian coordinates
+        for each node to do the calculation but the input data only has the
+        "Mesh2_node_x" and "Mesh2_node_y" in degree.
+
+        Note
+        ----
+        In the UXarray, we abide the UGRID convention and make sure the following attributes will always have its
+        corresponding units as stated below:
+
+        Mesh2_node_x
+         unit:  "degree_east" for longitude
+        Mesh2_node_y
+         unit:  "degrees_north" for latitude
+        Mesh2_node_z
+         unit:  "m"
+        Mesh2_node_cart_x
+         unit:  "m"
+        Mesh2_node_cart_y
+         unit:  "m"
+        Mesh2_node_cart_z
+         unit:  "m"
+        """
+
+        # Check if the cartesian coordinates are already populated
+        if "Mesh2_node_cart_x" in self.ds.keys():
+            return
+
+        # check for units and create Mesh2_node_cart_x/y/z set to self.ds
+        num_nodes = self.ds.Mesh2_node_x.size
+        node_cart_list_x = [0.0] * num_nodes
+        node_cart_list_y = [0.0] * num_nodes
+        node_cart_list_z = [0.0] * num_nodes
+        for i in range(num_nodes):
+            if "degree" in self.ds.Mesh2_node_x.units:
+                node = [
+                    np.deg2rad(self.ds["Mesh2_node_x"].values[i]),
+                    np.deg2rad(self.ds["Mesh2_node_y"].values[i])
+                ]  # [lon, lat]
+                node_cart = _convert_node_lonlat_rad_to_xyz(node)  # [x, y, z]
+                node_cart_list_x[i] = node_cart[0]
+                node_cart_list_y[i] = node_cart[1]
+                node_cart_list_z[i] = node_cart[2]
```

Comment:
The majority of this loop should be vectorizable, avoiding the need to iterate over every single node individually

Reply:
Thank you for your information! I wonder if you have any suggestions for the detailed vectorizations here? I know the `map` function and haven't tried it much.  Thank you for your help!

Reply:
You're welcome! The comments below this one should be a good starting point for an efficient vectorization. Essentially, instead of iterating over each node and indexing into arrays (which can be extremely slow in Python), we pass through our entire arrays into the Numpy functions. The following article is a good read! 

https://www.geeksforgeeks.org/vectorized-operations-in-numpy/